api/v1/services/TrackingService.js

```diff
@@ -0,0 +1,91 @@
+
+const TrackingFooter = "UniversalTracking";
+
+module.exports.createTrackingEvent = function (attributes) {
+    var trackingItem = TrackingItem.forge(attributes);
+
+    return trackingItem
+        .validate()
+        .catch(CheckitError, utils.errors.handleValidationError)
+        .then(function (validated) {
+            return TrackingItem.findByName(attributes.name);
+        })
+        .then(function (result) {
+            if (!_.isNull(result)) {
+                var message = "This event is already being tracked";
+                var source = "name";
+                throw new errors.InvalidParameterError(message, source);
+            }
+
+            return cache.getAsync('trackedEvent')
+        })
+        .then(function (result) {
+            if(!_.isNil(result)) {
+                return cache.ttlAsync('trackedEvent')
+                    .then(function (ttl) {
+                        var message = result + " is already being tracked at this time, time left: " + ttl.toHMS();
+                        var source = trackingItem.get('name');
+                        return _Promise.reject(new errors.InvalidTrackingStateError(message, source));
+                    });
+            }
+
+            return cache.multi()
+                .set('trackedEvent', trackingItem.get('name'))
+                .expire('trackedEvent', trackingItem.get('duration'))
+                .execAsync()
+                .then(function () {
+                    return trackingItem.save();
+                });;
+        })
+};
+
+module.exports.addEventParticipant = function(participantId) {
+    return cache.getAsync('trackedEvent')
+        .then(function (currentEvent) {
+            if(_.isNil(currentEvent)) {
+                var message = "No event is currently being tracked";
+                var source = "EventTracking";
+                throw new errors.InvalidTrackingStateError(message, source);
+            }
+
+            return cache.getAsync(participantId + TrackingFooter)
+                .then(function (result) {
+                    if(!_.isNil(result)) {
+                        var message = 'This attendee has already participated in ' + currentEvent + '!';
+                        var source = participantId;
+                        return _Promise.reject(new errors.InvalidParameterError(message, source));
+                    }
+
+                    return _Promise.resolve(currentEvent);
+                });
+        })
+        .then(function (currentTrackedEvent) {
+           return cache.ttlAsync('trackedEvent')
+                .then(function (ttl) {
+                    return cache.multi()
+                        .set(participantId + TrackingFooter, true)
+                        .expire(participantId + TrackingFooter, ttl)
+                        .execAsync()
+                        .then(function () {
+                            return TrackingItem.query().where('name', currentTrackedEvent).increment('count',1);
```

Comment:
The client doesn't need to wait for this to happen.

1) Please stop nesting `then`s like this. We have promises so that we don't have to do that -- make them linear.
2) Once you fix the nesting, this part of the chain should increment the count and immediately return `null` instead of the database promise (so that we don't have to wait for the database).